backend/src/api/routes/register.ts

```diff
@@ -0,0 +1,77 @@
+import express, { NextFunction } from 'express';
+import {
+  extractPayload,
+  Company,
+  Query,
+  CompanyUser,
+  instanceOfCompany,
+  instanceOfCompanyUser,
+  checkNotAuthenticated,
+} from '../../utils';
+import { executeTransaction, getAdminPoolClient, executeQuery } from '../../services/databasefunctions';
+import { logger } from '../../services/logger';
+import bcrypt from 'bcrypt';
+const app: express.Application = express();
+
+const newCompany = async (req: express.Request, res: express.Response, next: NextFunction): Promise<void | never> => {
+  try {
+    const payload = (await extractPayload(req)) as Record<string, any>;
+    if (instanceOfCompany(payload)) {
+      const { email, orgnum, username, salt, password, name }: Company = payload;
+      const passhash = await bcrypt.hash(password, 10);
+      logger.info(`Creating new company ${username}`);
+      const actions: Query = {
+        sqlStatement: 'INSERT INTO company(email,orgnum,username,salt,passhash,name) VALUES ($1,$2,$3,$4,$5,$6);',
+        data: [email, String(orgnum), username, salt, passhash, name],
```

Comment:
Is the username, salt and passhash still needed?

Reply:
The salt can be left out, we still need the passhash, but i'm uncertain about the username. 

I use it to generate users in the db, but since i wanted to avoid to much changing of the roles. But i could just rewrite it to only use email, both for the role and for the authentication, and just delete and remake roles/database entries if someone updates their email.

If you could give feedback on that thought i'll implement whatever you think is fitting shortly.